internal/armada/server/submit_to_log.go

```diff
@@ -245,6 +245,48 @@ func (srv *PulsarSubmitServer) SubmitJobs(ctx context.Context, req *api.JobSubmi
 }
 
 func (srv *PulsarSubmitServer) CancelJobs(ctx context.Context, req *api.JobCancelRequest) (*api.CancellationResult, error) {
+
+	// For requests specifying a job id, we need to check which queue the job is part of
+	// (see comments below).
+	// This must be done before checking authroization, since the auth check expects a queue.
+	if req.JobId != "" {
+		jobs, err := srv.SubmitServer.jobRepository.GetJobsByIds([]string{req.JobId})
+		if err != nil {
+			return nil, err
+		}
+		if len(jobs) != 1 { // Internal error; should never happen.
+			return nil, fmt.Errorf("expected 1 job result, but got %v", jobs)
+		}
+		queue := jobs[0].Job.GetQueue()
+		jobSetId := jobs[0].Job.GetJobSetId()
+
+		// We allow clients to submit requests only containing a job id.
+		// For these requests, we need to populate the queue and jobSetId fields.
+		if req.Queue == "" {
```

Comment:
shouldn't we always fill in the queue here?  Otherwise a malicious user could fill in the queue to whatever they want and they would effectively bypass our check.